sdk/tests/iot/common/test_az_iot_common.c

```diff
@@ -59,6 +59,17 @@ static void test_az_iot_message_properties_init_NULL_props_fails(void** state)
   ASSERT_PRECONDITION_CHECKED(az_iot_message_properties_init(NULL, test_span, 0));
 }
 
+static void test_az_iot_message_properties_init_user_set_params_too_many_written_fail(void** state)
+{
+  (void)state;
+
+  az_span test_span = az_span_create_from_str(TEST_KEY_VALUE_ONE);
+  az_iot_message_properties props;
+
+  ASSERT_PRECONDITION_CHECKED(
+      az_iot_message_properties_init(&props, test_span, az_span_size(test_span) + 1));
```

Comment:
While you are here, can you verify if a test exists for negative written length, and add one if it doesn't?